app/controllers/student_queues_controller.rb

```diff
@@ -24,8 +24,8 @@ def new
 
   def create
     student = Student.find(params[:id]) #after nesting student_queue routes, {:id => :student_id}
-    if student.student_queue.nil?
-      student.build_student_queue(:course => params[:course])
+    if student.student_queues.empty?
```

Comment:
A few questions, otherwise LGTM.
1) Are we allowing students to sign up for more than one thing within the same time frame? eg A student might have a weekly_appointment from 10am to 11am, and make a drop in request assuming the wait time is more than an hour. Currently, a student will only be allowed to make one type of appointment in a given time frame. The next question pretty much builds on this one.

2) What happens to a student after they have been seen by the tutor? Do they stay in the studentQueue? Assuming yesterday's discussion, the StudentQueue model will serve as a cache for the day and only transfer data to the history at the end of the day. If this is correct then then we will only be allowing students to have one dropin/weekly/scheduled session per day. Otherwise we should handle this more gracefully.

Additionally, looking at *student_queues#destroy*, if a student cancels, they won't be allowed to make any more appointments since they stay in queue and we ask *student.student_queues.empty?* to decided whether or not to put them in line.

Reply:
You're right. Right now the implementation doesn't support multiple requests. We can just remove the check and always create the request no matter what but that can lead to abuse. We should probably identify a limitation on the requests and then apply these changes next iteration.